features/features_test.go

```diff
@@ -378,59 +378,93 @@ func (s *featureSuite) TestAll(c *C) {
 	defer st.Unlock()
 	tr := config.NewTransaction(st)
 
+	fakeFeature := features.SnapdFeature(features.NumberOfFeatures())
+	is, why := fakeFeature.IsSupported()
+	c.Check(is, Equals, true)
+	c.Check(why, Equals, "")
+
+	restore := features.MockKnownFeaturesImpl(func() []features.SnapdFeature {
+		return []features.SnapdFeature{fakeFeature}
+	})
+	defer restore()
+
+	features.FeatureNames[fakeFeature] = "fake-feature"
+	defer func() {
+		delete(features.FeatureNames, fakeFeature)
+	}()
+
 	allFeaturesInfo := features.All(tr)
 
+	c.Assert(len(allFeaturesInfo), Equals, 1)
+
 	// Feature flags are included even if value unset
-	layoutsInfo, exists := allFeaturesInfo[features.Layouts.String()]
+	fakeFeatureInfo, exists := allFeaturesInfo[fakeFeature.String()]
 	c.Assert(exists, Equals, true)
 	// Feature flags are supported even if no callback defined.
-	c.Check(layoutsInfo.Supported, Equals, true)
+	c.Check(fakeFeatureInfo.Supported, Equals, true)
 	// Feature flags have a value even if unset.
-	c.Check(layoutsInfo.Enabled, Equals, true)
+	c.Check(fakeFeatureInfo.Enabled, Equals, false)
+
+	var fakeSupported bool
+	var fakeReason string
+	fakeCallback := func() (bool, string) {
+		return fakeSupported, fakeReason
+	}
+	features.FeaturesSupportedCallbacks[fakeFeature] = fakeCallback
+	defer func() {
+		delete(features.FeaturesSupportedCallbacks, fakeFeature)
+	}()
 
 	// Feature flags with defined supported callbacks work correctly.
 
 	// Callbacks which return false result in Supported: false
-	restore := systemd.MockSystemdVersion(229, nil)
-	defer restore()
+	fakeSupported = false
+	fakeReason = "foo"
 	allFeaturesInfo = features.All(tr)
-	quotaGroupsInfo, exists := allFeaturesInfo[features.QuotaGroups.String()]
+	c.Assert(len(allFeaturesInfo), Equals, 1)
+	fakeFeatureInfo, exists = allFeaturesInfo[fakeFeature.String()]
 	c.Assert(exists, Equals, true)
-	c.Check(quotaGroupsInfo.Supported, Equals, false)
-	c.Check(quotaGroupsInfo.UnsupportedReason, Matches, "systemd version 229 is too old.*")
-	c.Check(quotaGroupsInfo.Enabled, Equals, false)
+	c.Check(fakeFeatureInfo.Supported, Equals, false)
+	c.Check(fakeFeatureInfo.UnsupportedReason, Matches, fakeReason)
+	c.Check(fakeFeatureInfo.Enabled, Equals, false)
+
+	snapName, confName := fakeFeature.ConfigOption()
 
 	// Feature flags can be enabled but unsupported.
-	c.Assert(tr.Set("core", "experimental.quota-groups", "true"), IsNil)
+	c.Assert(tr.Set(snapName, confName, "true"), IsNil)
```

Comment:
FWIW it's harder to find out what is being passed around if it's hidden behind variable names, especially if you try to search the code with your favorite grep like tool. If you want to use variables here, then there there should be a line somewhere:
```
c.Check([]string{snapName, confName}, Equals, []string{"core", "experimental.fake-feature"})
```